### ISSUE TITLE
400 error on HTTP request to AirGradient

### ISSUE THREAD

**Issue by mafranklin** [MallocArray/airgradient_esphome#3]
Thank you for your work on this, I seem to having some problems sending the data to the AirGradient dashboard. I was using your code for this, but from the logs this error is returned:

`[12:10:25][W][http_request:093]: HTTP Request failed; URL: http://hw.airgradient.com/sensors/airgradient:XXXXXX/measures; Code: 400; Duration: 276 ms`

I was not sure if this was related to your description in the readme

> At this time the ChipID needed for adding to the AirGradient website is not displayed. May need to use Arduino sketch to get this ID then use ESPHome

Since the code you created seems to successfully pull the ChipID (at least as used by the SSID).

Apologies if this is something going wrong on my end / the AirGradient end, but any thoughts would be greatly appreciated!

**Comment by MallocArray** [comment]
I haven't checked the AirGradient site in a long time but I did see in my ESPHome logs that I'm having an error as well, although mine doesn't say Code 400 but `Error: connection failed;`

I opened a thread on the AirGradient forum to see what we can find.  Mine stopped talking to the API on Feb 5th

**Comment by MallocArray** [comment]
In my case, at some point I had commented out my DNS servers while keeping the static IP and that kept it from resolving the hostname.  Adding the DNS server back in allowed mine to connect again.  
If you have DHCP setup, that shouldn't be the issue

I was able to confirm I could connect by using Postman on my PC to try to send a POST to http://hw.airgradient.com/sensors/airgradient:xxxxxx/measures with a json body of:
```
{
    "pm02": 1
}
```

This was successful which let me know my network wasn't blocking connection, so I looked closer at the wifi config of my ESPHome device and found the culprit.

**Comment by mafranklin** [comment]
Thanks for looking into this. I will give this a go now. I saw your forum thread over on AirGradient as well, so have also adjusted the interval for sending data to 90s to avoid the 429 error.

**Comment by mafranklin** [comment]
Just to confirm that it is all working now as expected after defining the DNS servers. Thanks again!

**Comment by mcgibbaa** [comment]
@MallocArray I had a similar problem, may be worth adding to the readme or pulling this somewhere more visible?
 For context:
I defined a static IP/gateway/subnet in my ESPHome yaml, and the data was not being sent to AirGradient. 
I use PiHole, so once I added the DNS server definition to point to the PiHole it started working again.
I'm guessing this may be true for other ESPHome devices but since I don't have any other ESPHome devices pointing to a cloud server haven't encountered this before!
